File: TwinTrades/TwinTrades/Models/Trade.swift
import Foundation

enum TradeAction: String, Codable {
    case buy
    case sell
}

enum TradeStatus: String, Codable {
    case pending
    case filled
    case cancelled
    case failed
}

struct Trade: Codable, Identifiable {
    let id: Int
    let ticker: String
    let action: TradeAction
    let quantity: Decimal
    let price: Decimal?
    let status: TradeStatus
    let brokerOrderId: String?
    let executedAt: Date?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, ticker, action, quantity, price, status
        case brokerOrderId = "broker_order_id"
        case executedAt = "executed_at"
        case createdAt = "created_at"
    }
}

struct TradeHistoryResponse: Codable {
    let trades: [Trade]
    let total: Int
    let page: Int
    let pageSize: Int

    enum CodingKeys: String, CodingKey {
        case trades, total, page
        case pageSize = "page_size"
    }
}

File: frontend/models/Trade.ts
export type TradeAction = 'buy' | 'sell';
export type TradeStatus = 'pending' | 'filled' | 'cancelled' | 'failed';

export interface Trade {
  id: number;
  ticker: string;
  action: TradeAction;
  quantity: string;
  price: string | null;
  status: TradeStatus;
  broker_order_id: string | null;
  executed_at: string | null;
  created_at: string;
}

export interface TradeHistoryResponse {
  trades: Trade[];
  total: number;
  page: number;
  page_size: number;
}
